### DISCUSSION TITLE
Using scss with in nextjs 9.3

### DISCUSSION THREAD

**Discussion by zackfalcon** [vercel/next.js#12510] [Help] [answered]
Hi, today i'm actually using nextjs to refactor an react app into nextjs

**Question 1 :**
 So i already had my styles scss prepared, no problem i imported them, but actually, i need to comment every line that use define style on **html tag  [a, body, ...]** if not, i have this type of error and i can't run my app :

`CssSyntax error: Selector "a" is not pure (pure selectors must contain at least one local class or id) (37:0)

  35 | // }
  36 | 
> 37 | a {
     |             ^
  38 |   text-decoration: none !important;
  39 |   &:focus {`

`CssSyntax error: Selector "h1, h2, h3" is not pure (pure selectors must contain at least one local class or id) (33:0)

  31 | }
  32 | 
> 33 | h1, h2, h3 {
     |                      ^
  34 |   margin: 0px;
  35 | }`

**Question 2:**

Also i wanted to know what is this is a good configuration for my scss cause i saw that next 9.3 add : 
`◆ Built-In Sass Imports for Global Styles
◆ Built-In Sass Modules for Component-Level Styles`

But for me it's actually working pretty great with only this config in my **next.config.js** it work on _dev_ and _production_ on heroku :

`module.exports = {
  experimental: { scss: true },
};
`
But every time i run my app i have this message : 
Don't understand why : **Warning: You have enabled experimental feature(s).
Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use them at your own risk.**
 
 **Question 3** 

So as i said, it's actually working pretty great instead of one thing again about my style when i actually wont to apply my className in a component or anywhere i need to use this type of method : 

my import looks like this ``import styles from './HomeMain.module.scss';```

if i want to apply a style i need to do it like this : `        <div className={styles["home-select-menuList"]}>
` instead of the old method using css module in react `        <div className={"home-select-menuList"}>
` 
You will say okay its works why do u want to change this but actually if u want to apply two class on the same element you can do it only like this : `<div className={``${styles["home-select-menuList"] } ${styles["home-select-menuList--clicked"]}``}>

So it's really boring and i can't use the famous library classnames easily how can i solve it using the old way like css module in react ! 

Thanks a lot ! And great work for the 9.3 V !

**Comment by jacktan165** [comment, marked as answer]
Put all your global CSS in `/styles` and import it in `/pages/_app.tsx`.

You are doing this `<div className={styles["home-select-menuList"]}>` is because you are using CSS Modules. 
If you don't like it, just use https://github.com/JedWatson/classnames or make your own little util function which accepts an arbitrary number of strings, and join them with an empty space. 

`const cn = (...classNames: string[]): string => classNames.join(" ");`

That's how I do it.

**Reply by zackfalcon** [reply]
Hey ! THanks for your answer !

(Put all your global CSS in /styles and import it in /pages/_app.tsx.) 

I already have done this, but i have this type of issu 
``CssSyntax error: Selector "a" is not pure (pure selectors must contain at least one local class or id) (37:0)

35 | // }
36 |
37 | a {
| ^
38 | text-decoration: none !important;
39 | &:focus {``


Thanks for the tips ! function !

**Reply by jacktan165** [reply]
I tried it on my machine and it works fine.

Did you do something like this?

`import "../styles/index.scss";`

**Reply by jacktan165** [reply]
Or did you do anything to your `next.config.js` file?

**Reply by zackfalcon** [reply]
YOu can see my architecture here and _app file with nextconfig : https://image.noelshack.com/fichiers/2020/19/2/1588683710-capture-d-ecran-de-2020-05-05-09-01-23.png

**Reply by jacktan165** [reply]
Move your `index.scss` inside `/styles`, see if that works.
You don't need this `module.exports = { experimental: { scss: true }, }` btw...

**Reply by zackfalcon** [reply]
JUst changed it it wont works here is the file and error : https://image.noelshack.com/fichiers/2020/19/2/1588685723-capture-d-ecran-de-2020-05-05-09-35-16.png

**Reply by zackfalcon** [reply]
But it works actually for the website like before, but i have this error for every html tag again

**Comment by naveen-bharathi** [comment]
@zackfalcon you are receiving this error because you are using html tags directly instead of classnames or ids in a file extension that is probably **[filename].module.(css | scss | sass)**

File extensions with ***.module.(css | scss | sass)** are css modules and they can only target elements using classnames or ids and not using tag names. Although this is possible in other frameworks like create-react-app, it is not possible in next-js.

My suggestion is using these html selector css in a separate file that doesn't contain the **'.module'** in it. 
Example: **[filename].(css | scss | sass)**  -->  **styles.scss**


And instead of importing like this
```js
import styles from './styles.module.scss';
```
import like this
```js
import './styles.scss';
```

This will not through any errors.

**Reply by Rujal-Duwal** [reply]
Then it will be global CSS.  I need such CSS in specific components only. Please support me as I am new with Nextjs.

**Comment by bumblebee338** [comment]
I have this file _typography.scss in styles/_typography.scss in Nextjs

<img width="454" alt="Screenshot 2023-06-15 at 4 49 17 PM" src="https://github.com/vercel/next.js/assets/55170754/b7dcb62f-4c3f-48fe-965e-62154a969a7d">

there are only variable in variables.scss so, I didn't change that

<img width="791" alt="Screenshot 2023-06-15 at 4 29 08 PM" src="https://github.com/vercel/next.js/assets/55170754/8481b50f-6b5a-444f-88b9-bca3aa5e7541">

I am getting this error for pure selector, but typography file is just to import (The _ (underscore) is a partial for scss. That means the stylesheet its going to be imported (@import) to a main stylesheet) I cannot give any classname or ID in that file, if i have to give any className there then i have to give that every component

``` <main className={`${styles.container} ${styles.typo}`}>```

<img width="716" alt="Screenshot 2023-06-15 at 4 29 56 PM" src="https://github.com/vercel/next.js/assets/55170754/b040aad3-1c09-4377-8272-934e5d828508">

Since keeping it scss is global css it can only be imported in _app.js

**Is there any other solution to this ?**
